build_tools/travis/flake8_diff.sh

```diff
@@ -120,14 +120,25 @@ echo '--------------------------------------------------------------------------
 # uses git 1.8.
 # We need the following command to exit with 0 hence the echo in case
 # there is no match
-MODIFIED_FILES=$(git diff --name-only $COMMIT_RANGE | grep -v 'sklearn/externals' | \
-                     grep -v 'doc/sphinxext/sphinx_gallery' || echo "no_match")
+MODIFIED_FILES="$(git diff --name-only $COMMIT_RANGE | grep -v 'sklearn/externals' | \
+                     grep -v 'doc/sphinxext/sphinx_gallery' || echo "no_match")"
+
+check_files() {
+	files="$1"
+	options="$2"
+	echo "$files"
+	echo $options
+	git diff --unified=0 $COMMIT -- $files | flake8 --diff --show-source $options
+}
 
 if [[ "$MODIFIED_FILES" == "no_match" ]]; then
     echo "No file outside sklearn/externals and doc/sphinxext/sphinx_gallery has been modified"
 else
     # Conservative approach: diff without context so that code that
     # was not changed does not create failures
-    git diff --unified=0 $COMMIT -- $MODIFIED_FILES | flake8 --diff --show-source
+	check_files "$(echo "$MODIFIED_FILES" | grep -v ^examples)"
+    # Examples are allowed to not have imports at top of file due to print(__doc__)
```

Comment:
I would remove due to `print(__doc__)`. This is not the main reason, this happens most of the time because examples are written in a "interactive mode" where you only import stuff when you need it rather than at the top of the file.